spring-vault-core/src/main/java/org/springframework/vault/core/VaultTransitOperations.java

```diff
@@ -222,4 +222,67 @@
 	 * @see #rotate(String)
 	 */
 	String rewrap(String keyName, String ciphertext, VaultTransitContext transitContext);
+
+	/**
+	 * Generate HMAC digest of given data.
+	 *
+	 * @param keyName must not be empty or {@literal null}.
+	 * @param plainText must not be empty or {@literal null}.
+	 * @return the digest of given data using sha2-256 hash algorithm and the named key.
+	 */
+	String generateHmac(String keyName, String plainText);
+
+	/**
+	 * Generate HMAC digest of given data.
+	 *
+	 * @param keyName must not be empty or {@literal null}.
+	 * @param plainText must not be empty or {@literal null}.
+	 * @param algorithm Specify the algorithm to be used for signing. Supported algorithms are:
+	 *                  sha2-224, sha2-256, sha2-384, sha2-512
+	 * @return the digest of given data using the specified hash hash algorithm and the named key.
+	 */
+	String generateHmac(String keyName, String plainText, String algorithm);
```

Comment:
It would make sense to provide a `VaultHmacRequest` object that encapsulates `PlainText`, `algorithm` and possibly other options that apply to the HMAC operation.